src/transformers/modeling_utils.py

```diff
@@ -1907,6 +1907,33 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             torch_dtype (`str` or `torch.dtype`, *optional*):
                 Override the default `torch.dtype` and load the model under this dtype. If `"auto"` is passed the dtype
                 will be automatically derived from the model's weights.
+
+                Here is how `torch_dtype` defines in which `dtype` the model gets loaded:
+
+                1. `torch_dtype=torch.float16` - (or `torch.bfloat16` or `torch.float`) load in a
+                  specified `dtype`, ignoring the model's `config.torch_dtype` if one exists
+
+                2. `torch_dtype=config.torch_dtype` - use the `dtype` specified in the config (similar to
+                  the first case
```

Comment:
If you want to also implement the `"config"` choice, it would be better than somehow extracting it manually.

Reply:
certainly, let me rework it then first and then I will update the doc.

Reply:
I think 1 is good: it's a value error easy to fix.

Reply:
plus @ydshieh makes a good point, that if the gpu isn't supporting bf16, then should we assert?

https://huggingface.slack.com/archives/C01N44FJDHT/p1675887341538559

Reply:
but wait, when the model is loaded we don't know if it's going to be used on gpu, it's created on `cpu`, so probably we shouldn't make any assumptions about it at this point and use bf16 if told so.

Reply:
all implemented.

Reply:
ok, "config" got folded into "auto" - all done and is ready for your review at your convenience, Sylvain.